docs/oci-storage.md

```diff
@@ -1,40 +1,149 @@
-    // expects a generic secret with username and password keys set.
-    authSecretName: oci-secret
-    // basicHTTP allows Fleet to use basic http connections to communicate
-    // with the registry (defaults to false)
-    basicHTTP: false
-    // insecureSkipTLS allows connections to the OCI registry
-    // without certs (defaults to false)
-    insecureSkipTLS: false
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  reference: <base64-encoded-registry-url>
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+# Optional
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
 ```
+:::note
+Only the reference field is required. All other fields are optional. 
+:::
+
+If no username or password is specified, Fleet accesses the registry without authentication. You should follow these best practices:
+
+* Use read-only `agentUsername` and `agentPassword` credentials for agents to enhance security.  
+  * However, if you don’t set these credentials, the agent uses user credentials with read/write permissions.  
+* Insecure TLS defaults to `false`, and basic http is disabled by default.   
+  * Fleet allows these flags for development and testing purposes, but they should never be used in production.  
+  * If you use these tags, you expose your app to security vulnerabilities, and put cluster workloads, and credentials at a risk of tampering.  
+* Changing the secret does not update the deployment. The new storage registry is used only after the next Git update or when you trigger a force update.
+
+:::note
+Fleet checks for the integrity of OCI artifacts and Fleet tags OCI artifact as Latest
+:::
+
+### Fleet Example
+
+Consider the GitRepo file for fleet-examples repository. 
+
+```yaml
+apiVersion: fleet.cattle.io/v1alpha1
+kind: GitRepo
+metadata:
+  name: frontend-oci
+  namespace: fleet-local
+spec:
+  repo: https://github.com/your-org/fleet-oci-example.git
+  branch: main
+  paths:
+    - ./frontend
+  ociRegistrySecret: ocistorage
+```
+
+To define OCI Secret, you can create an oci-secret.yaml file
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
+```
+
+Run `kubectl apply -f secrets/oci-secret.yaml` before applying the GitRepo.
+
+Or you can use `kubectl` command to create the `ocistorage` secret using unencoded text. Kubernetes converts them to base64 encoded for storing the secret.
+
```

Comment:
I would mention that the `reference` field is a URL without prefixes. 
For example, no `oci://` prefix should be added.